pkg/controller/inferenceservice/resources/istio/virtualservice.go

```diff
@@ -266,6 +266,14 @@ func (r *VirtualServiceBuilder) CreateVirtualService(isvc *v1alpha2.InferenceSer
 	// extract the virtual service hostname from the predictor hostname
 	serviceURL := fmt.Sprintf("%s://%s", "http", serviceHostname)
 
+	serviceInternalHostName := network.GetServiceHostname(isvc.Name, isvc.Namespace)
+	hosts := []string{
+		serviceHostname,
```

Comment:
I think the default host should be `serviceInternalHostName`, there is another case we should take in account here is that cluster local service can be also configured at service level with label `serving.knative.dev/visibility=cluster-local` according to [the doc](https://knative.dev/docs/serving/cluster-local-route). So I'd suggest the following change

```
	serviceInternalHostName := network.GetServiceHostname(isvc.Name, isvc.Namespace)
	hosts := []string{
		serviceInternalHostName,
        }
        isClusterLocal := false
        if val, ok := isvc.metadata.labels["serving.knative.dev/visibility"]; ok {
              if val == "cluster-local" {
                     isClusterLocal = true
              } 
        }
        if serviceHostname != serviceInternalHostName && !isClusterLocal {
		hosts = append(hosts, serviceHostname)
	}
```

Reply:
Thanks for the pointers, I totally missed the possibility to use labels for that. Updated the code and tests.